src/App.tsx

```diff
@@ -1,61 +1,118 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+let lastId = Math.max(...todos.map(({ id }) => id));
+
+export const App: React.FC = () => {
+  const [newTodos, setNewTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const getNewTodo = () => {
+    const newId = lastId + 1;
+
+    lastId += 1;
```

Comment:
Nice job!
Approve.
But retnink about this place.
I suggest to count lastId in this function and after that you will not think about increasing lastId.
In bigger apps it can be potential unrespected errors.